weaver/processes/wps_package.py

```diff
@@ -1962,9 +1979,34 @@ def _handler(self, request, response):
                 # ADES execute the cwl locally
                 loading_context = None
 
-            wps_out_dir_prefix = os.path.join(get_wps_output_dir(settings), "tmp")
-            runtime_args = {"no_read_only": True, "outdir": self.workdir, "tmp_outdir_prefix": wps_out_dir_prefix}
-            runtime_context = RuntimeContext(kwargs=runtime_args)
+            self.update_effective_user()
+            self.update_requirements()
+
+            # note:
+            #   Parameter 'weaver.wps_workdir' is the base-dir where sub-dir per application packages will be generated.
+            #   Parameter 'self.workdir' is the actual location PyWPS reserved for this process (already with sub-dir).
+            #   If no 'weaver.workdir' was provided, reuse PyWps parent workdir since we got access to it.
```

Comment:
If we reuse PyWps parent workdir, it means that cwl app packages private dir will exist along proces dir? We will then expose those private directories on the web? 
Why not requiring such a parameter or creating one under /tmp ?

Reply:
The pywps/cwltool directories are at the same level, but the workdir is not on the wps-outputs path. It is some `/tmp/whatever/pywps_process_<random>` location with `/tmp/whatever` configured using `weaver.wps_workdir`. The results generated by this are automatically moved to path corresponding to `{weaver.wps_output_dir}/{job-id}` when the process completes.